doc/website/getting-started/overview.md

```diff
@@ -11,637 +10,290 @@ runtime and create _publishers_ and _subscribers_. The publishers send data of a
-
-        auto result = subscriber->take();
-
-        if(!result.has_error()) 
-        {
-            auto& maybeSample = result.value();
-            if (maybeSample.has_value())
-            {
-                auto& sample = maybeSample.value();
-                void* chunk = sample.get();
-
-                //interpret and process the data
-                const CounterTopic* counter = reinterpret_cast<const CounterTopic*>(chunk);
-            }
-            else
-            {
-                // we received no data
-            }
-        } else {
-            iox::popo::ChunkReceiveResult& error = result.get_error();
-            //handle the error
-        }
-    }
-```
+For more information on how to use the WaitSet see our
+[WaitSet examples](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_examples/waitset).
 
-### Shutdown
+### Listener
 
-Once we are done sending, we call ``stopOffer`` at the publisher.
+The Listener can be used to connect custom callbacks to user-defined events. Unlike the WaitSet, it reacts to those 
+events by executing the connected custom callbacks in a background thread. 
 
-```cpp
-publisher.stopOffer();
-```
+!!! note 
+    The Listener is completely thread-safe but please be aware that most of the objects which can be attached to the 
+    Listener are not thread-safe! This means either the object is handled solely by the Listener, which should be the 
+    most common use case, or the user has to ensure the thread safety with other means like encapsulating the object in 
+    a thread-safe class.
 
-Similarly the subscriber can ``unsubscribe`` to stop receiving any data.
+One use case could be that one creates a Listener and attaches multiple subscribers. Every time new data is available, 
+the corresponding connected callback will be executed, e.g. print something to the console or calculate something.
 
-```cpp
-subscriber.unsubscribe();
-```
+Like the WaitSet, the Listener uses the reactor pattern.
 
-Both will also be called in the respective destructor if needed (i.e. if the publisher is still offering or the
-subscriber is still subscribed).
+For more information about the Listener see our 
+[callbacks example](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_examples/callbacks).
 
-### Running an iceoryx system
+## API
 
-Now that we have applications capable of sending and receiving data, we can run the complete iceoryx system.
+The API is offered in two languages, C++ and C. Detailed information can be found in the 
+[C++ example](../examples/icedelivery/) and 
+[C example](../examples/icedelivery_in_c/).
 
-First we need to start RouDi.
+Many parts of the C++ API follow a functional programming approach which is less error-prone. This requires using 
+the monadic types ``cxx::expected`` and ``cxx::optional`` which are introduced 
+[here](http://iceoryx.io/advanced/usage-guide/how-optional-and-error-values-are-returned-in-iceoryx/).
```

Comment:
Could you change this link to `../advanced/how-optional-and-error-values-are-returned-in-iceoryx.md`?